.github/workflows/rust-clippy.yml

```diff
@@ -0,0 +1,55 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# rust-clippy is a tool that runs a bunch of lints to catch common
+# mistakes in your Rust code and help improve your Rust code.
+# More details at https://github.com/rust-lang/rust-clippy
+# and https://rust-lang.github.io/rust-clippy/
+
+name: rust-clippy analyze
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '37 12 * * 1'
```

Comment:
Feels like this should just be done on commit to main through a merge. Is this github being sneaky about credit usage ?

Reply:
You're correct, this was a super old PR so not sure of the context of this cron, seems out of place. Not yet merging this since Clippy is a bit much when trying to move quick.